### DISCUSSION TITLE
Shipping Method Types

### DISCUSSION THREAD

**Discussion by brandonkelly** [craftcms/commerce#2281] [Ideas]
Plugins should be adding new shipping methods the same way they add new gateway types, field types, etc.. So all shipping methods become managed by Commerce, not just the "custom" ones.

- [ ] Figure out what to do with the Rules settings, which are currently only used by Custom shipping methods.

**Comment by lukeholder** [comment]
Could be dupe of #70

**Comment by sjcallender** [comment]
Regarding the shipping rules, it would be helpful if they could be managed separately from the methods and applied to multiple methods without recreating them for each method.

**Comment by lukeholder** [comment]
Add type column to shipping methods, and give them settings.
